DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -28,20 +37,33 @@ namespace edmNew {
 
   namespace dslv {
     template< typename T> class LazyGetter;
+
   }
 
   /* transient component of DetSetVector
-   * for pure conviniency of dictioanary declaration
+   * for pure conviniency of dictionary declaration
    */
   namespace dstvdetails {
+
+    void errorFilling();
+    void notSafe();
+    void errorIdExists(det_id_type iid);
+    void throw_range(det_id_type iid);
+
     struct DetSetVectorTrans {
       DetSetVectorTrans(): filling(false){}
-      bool filling;
+      DetSetVectorTrans& operator=(const DetSetVectorTrans&) = delete;
+      DetSetVectorTrans(const DetSetVectorTrans&) = delete;
+      DetSetVectorTrans(DetSetVectorTrans&&) = default;
+      DetSetVectorTrans& operator=(DetSetVectorTrans&&) = default;
+      mutable std::atomic<bool> filling;
       boost::any getter;
 
 
       void swap(DetSetVectorTrans& rh) {
-	std::swap(filling,rh.filling);
+	// better no one is filling...
+        assert(filling==false); assert(rh.filling==false);	
```

Comment:
The test is nice, but we are only trying to enforce `const thread safety` and this is not a const function. Something to keep in mind for the future.